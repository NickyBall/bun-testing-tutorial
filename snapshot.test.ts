import { expect, test } from "bun:test";

function generateComplexObject(name: string, age: number) {
  return {
    name,
    age,
    createdAt: new Date(),
    preferences: {
      theme: "dark",
      notifications: true
    }
  };
}

test("generateComplexObject snapshot", () => {
  const object = generateComplexObject("John Doe", 30);

  // The first time this test runs, it will create a snapshot.
  // On subsequent runs, it will compare the output to the snapshot.
  expect(object).toMatchSnapshot({
    createdAt: expect.any(Date) // Ignore the exact date value
  });
});