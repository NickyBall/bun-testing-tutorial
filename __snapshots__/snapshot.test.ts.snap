// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`generateComplexObject snapshot 1`] = `
{
  "age": 30,
  "createdAt": Any<Date>,
  "name": "John Doe",
  "preferences": {
    "notifications": true,
    "theme": "dark",
  },
}
`;
